src/Configuration.php

```diff
@@ -9,42 +9,29 @@
 class Configuration
 {
     private $layers;
-
     private $paths;
-
     private $exclude_files;
-
     private $ruleset;
 
     public static function fromArray(array $arr)
     {
-        $options = (new OptionsResolver())->setRequired([
-            'layers',
-            'paths',
-            'ruleset',
-        ])
-        ->setDefault('exclude_files', [])
-        ->addAllowedTypes('layers', 'array')
-        ->addAllowedTypes('paths', 'array')
-        ->addAllowedTypes('exclude_files', ['array', 'null'])
-        ->addAllowedTypes('ruleset', 'array')
-        ->resolve($arr);
+        $options = self::resolveOptions($arr);
```

Comment:
this should be part of the fromArray method, it's just about validating the functions input.

Reply:
This logic/method should be moved to `ConfigurationLoader` or a Factory.
The Configuration should not have a dependency on `OptionResolver`.